File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/domain/model/JwtUserDetails.java
package com.windea.demo.cloudcollect.core.domain.model;

import com.windea.demo.cloudcollect.core.domain.entity.User;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;

import static org.springframework.security.core.authority.AuthorityUtils.createAuthorityList;

/**
 * Jwt用户详情。用于安全验证。
 */
public class JwtUserDetails implements UserDetails {
	private static final long serialVersionUID = -8990133011806369007L;

	private final User delegateUser;

	public JwtUserDetails(User delegateUser) {
		this.delegateUser = delegateUser;
	}


	public User getDelegateUser() {
		return delegateUser;
	}

	@Override
	public String getUsername() {
		return delegateUser.getUsername();
	}

	@Override
	public String getPassword() {
		return delegateUser.getPassword();
	}

	@Override
	public Collection<? extends GrantedAuthority> getAuthorities() {
		return createAuthorityList(delegateUser.getRole().toString());
	}

	@Override
	public boolean isAccountNonExpired() {
		return true;
	}

	@Override
	public boolean isAccountNonLocked() {
		return true;
	}

	@Override
	public boolean isCredentialsNonExpired() {
		return true;
	}

	@Override
	public boolean isEnabled() {
		return delegateUser.getActivateStatus();
	}
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/domain/response/JwtUserDetails.kt
package com.windea.demo.cloudcollect.core.domain.response

import com.windea.demo.cloudcollect.core.domain.entity.User
import org.springframework.security.core.*
import org.springframework.security.core.authority.*
import org.springframework.security.core.userdetails.*

/**Jwt用户详情。用于安全验证。*/
data class JwtUserDetails(
	val delegateUser: User
) : UserDetails {
	override fun getUsername(): String {
		return delegateUser.username
	}
	
	override fun getPassword(): String {
		return delegateUser.password
	}
	
	override fun getAuthorities(): Collection<GrantedAuthority> {
		return AuthorityUtils.createAuthorityList(delegateUser.role.toString())
	}
	
	override fun isAccountNonExpired(): Boolean {
		return true
	}
	
	override fun isAccountNonLocked(): Boolean {
		return true
	}
	
	override fun isCredentialsNonExpired(): Boolean {
		return true
	}
	
	override fun isEnabled(): Boolean {
		return delegateUser.isActivated
	}
}

